### PR TITLE
[4.0] Rename Fields - Subfields plugin to Fields - Subform plugin

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
With PR https://github.com/joomla/joomla-cms/pull/32864,  @brianteeman made some changes to language files to change subfields plugin to subform plugin. However, the actually plugin code is still uses old name **subfields**. This PR changes the plugin code to have it completely renamed from **subfields** to **subform** 

### Testing Instructions
1. Install Joomla 4 nightly build. Or if you have Joomla 4 installation already, please update that installation to Joomla 4 nightly build
2. Access to Content -> Fields, create at least one **subfields** custom field. For testing purpose, maybe you can create two text custom fields (these fields should have Categories set to None) and one subfields like in the attached screenshot (the subfields could have two text fields in it's Fields setting)

- List of Fields
subfields

- Subfields settings

subfields_setting

3. Create a new article, go to Fields tab and enter data for that subfields custom field for the article

4. Download update package for this PR at https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33200/downloads/42161/Joomla_4.0.0-beta8-dev+pr.33200-Development-Update_Package.zip , go to System -> Update -> Joomla, install that update package

5. Now, access to Content -> Fields again, confirm that the **subfields** which you created before is now become **subform**

subfields_become_subform

6. Edit that subform custom field (before it was subfields), confirm that **Type** now become **Subform (Subform)**. Also, the Fields setting still same as before
type_changed

7. Edit the article, go to Fields tab, confirm that the data which you entered for the custom fields before is still there.

8. View the article from frontend (maybe make it featured so that it is displayed at homepage by default) and confirm that the data you entered for the custom fields is being displayed properly

article

### CODE REVIEW

**Comment by richard67** [comment]
Several mistakes in SQL found, made a PR. See https://github.com/joomdonation/joomla-cms/pull/1 .

**Comment by joomdonation** [comment]
I updated testing instructions. This PR is now ready for testing. Please help testing it.

**Comment by Quy** [comment]
Subforms should not be listed in subforms.

Rename/update?
> \administrator\components\com_fields\src\Field\SubfieldstypeField.php

**Comment by joomdonation** [comment]
@Quy I thought about that before and decided that it is not necessary to change. We have Subform. That Subform contains some fields, so the fields which is used inside that Subform could be called as Subfields. 

That field type is used to list all fields which can be used in Subform, so leave the field type as how it was before is OK, I think

**Comment by joomdonation** [comment]
@Quy Or maybe we should rename it to **SubfieldsField.php**. Guess type is not needed in the field type/ filename/classname. I can make the change if needed.

**Comment by joomdonation** [comment]
So I think Subfields is a better name compare to Subfieldstype and made the change.

File: plugins/fields/subform/subform.php [hunk 1]
```diff
@@ -234,11 +234,11 @@ public function onCustomFieldsPrepareField($context, $item, $field)
 			}
 
 			// Store all the sub fields of this row
-			$subfields_rows[] = $row_subfields;
+			$subform_rows[] = $row_subfields;
 		}
 
 		// Store all the rows and their corresponding sub fields in $field->subfields_rows
```
Comment (Quy) [inline plugins/fields/subform/subform.php]:
```suggestion
		// Store all the rows and their corresponding sub fields in $field->subform_rows
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 1a398f4673f72112852e6017dc8c0d5d3778a3ea<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33200">issues.joomla.org/tracker/joomla-cms/33200</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 1a398f4673f72112852e6017dc8c0d5d3778a3ea<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33200">issues.joomla.org/tracker/joomla-cms/33200</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33200">issues.joomla.org/tracker/joomla-cms/33200</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**